examples/nucleo_f401re/adc_dma/main.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2023, Zühlke (Austria) gmbh
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// This Example uses adc to sample 2 channels in scan mode 50 times every 0.5s and store the value
+// in a buffer via dma. The adc is also set to be triggered by Timer1 compare Event, that is set in
+// one pulse mode with a repetition count set. For more information on the timer please refer to the
+// timer register count example. In this configuration, the adc sampling process is started by the
+// cpu but handled completely by peripherals clearing CPU time for other tasks.
+
+#include <modm/architecture/interface/interrupt.hpp>
+#include <modm/board.hpp>
+#include <modm/platform.hpp>
+
+#include "AdcDma.hpp"
+#include "TimerHandler.hpp"
+
+using namespace Board;
+using namespace modm::platform;
+using Adc1Dma = AdcDma<Adc1, Dma2::Channel0>;
+
+std::array<uint16_t, 100> adc_results;
+volatile bool dma_completed = false;
+
+void
+completedCallback(void)
+{
+	LedD13::toggle();
+	MODM_LOG_DEBUG << "DMA callback\r" << modm::endl;
+	dma_completed = true;
+}
+
+int
+main()
+{
+	Board::initialize();
+	Adc1::initialize<SystemClock>();
+	Dma2::enable();
+
+	// Use the logging streams to print some messages.
+	// Change MODM_LOG_LEVEL above to enable or disable these messages
+	MODM_LOG_DEBUG << "debug" << modm::endl;
+	MODM_LOG_INFO << "info" << modm::endl;
+	MODM_LOG_WARNING << "warning" << modm::endl;
+	MODM_LOG_ERROR << "error" << modm::endl;
+
+	LedD13::setOutput();
+	LedD13::reset();
+	advancedTimerConfig<Timer1>(adc_results.size() / 2 - 1);
+	Adc1Dma::initialize<SystemClock>((uintptr_t)(&adc_results[0]), adc_results.size());
```

Comment:
maybe group the 2 Adc1Dma calls together if possible

maybe add a few comments for each step?